syntax = "proto3";

package material;

message OptionalDouble {
    bool is_defined = 1;
    double value = 2;
}

message Material {
    string material_id = 1;

    OptionalDouble band_gap = 2;

    uint32 atoms_of_element_1 = 1001;
    uint32 atoms_of_element_2 = 1002;
    uint32 atoms_of_element_3 = 1003;
    uint32 atoms_of_element_4 = 1004;
    uint32 atoms_of_element_5 = 1005;
    uint32 atoms_of_element_6 = 1006;
    uint32 atoms_of_element_7 = 1007;
    uint32 atoms_of_element_8 = 1008;
    uint32 atoms_of_element_9 = 1009;
    uint32 atoms_of_element_10 = 1010;
    uint32 atoms_of_element_11 = 1011;
    uint32 atoms_of_element_12 = 1012;
    uint32 atoms_of_element_13 = 1013;
    uint32 atoms_of_element_14 = 1014;
    uint32 atoms_of_element_15 = 1015;
    uint32 atoms_of_element_16 = 1016;
    uint32 atoms_of_element_17 = 1017;
    uint32 atoms_of_element_18 = 1018;
    uint32 atoms_of_element_19 = 1019;
    uint32 atoms_of_element_20 = 1020;
    uint32 atoms_of_element_21 = 1021;
    uint32 atoms_of_element_22 = 1022;
    uint32 atoms_of_element_23 = 1023;
    uint32 atoms_of_element_24 = 1024;
    uint32 atoms_of_element_25 = 1025;
    uint32 atoms_of_element_26 = 1026;
    uint32 atoms_of_element_27 = 1027;
    uint32 atoms_of_element_28 = 1028;
    uint32 atoms_of_element_29 = 1029;
    uint32 atoms_of_element_30 = 1030;
    uint32 atoms_of_element_31 = 1031;
    uint32 atoms_of_element_32 = 1032;
    uint32 atoms_of_element_33 = 1033;
    uint32 atoms_of_element_34 = 1034;
    uint32 atoms_of_element_35 = 1035;
    uint32 atoms_of_element_36 = 1036;
    uint32 atoms_of_element_37 = 1037;
    uint32 atoms_of_element_38 = 1038;
    uint32 atoms_of_element_39 = 1039;
    uint32 atoms_of_element_40 = 1040;
    uint32 atoms_of_element_41 = 1041;
    uint32 atoms_of_element_42 = 1042;
    uint32 atoms_of_element_43 = 1043;
    uint32 atoms_of_element_44 = 1044;
    uint32 atoms_of_element_45 = 1045;
    uint32 atoms_of_element_46 = 1046;
    uint32 atoms_of_element_47 = 1047;
    uint32 atoms_of_element_48 = 1048;
    uint32 atoms_of_element_49 = 1049;
    uint32 atoms_of_element_50 = 1050;
    uint32 atoms_of_element_51 = 1051;
    uint32 atoms_of_element_52 = 1052;
    uint32 atoms_of_element_53 = 1053;
    uint32 atoms_of_element_54 = 1054;
    uint32 atoms_of_element_55 = 1055;
    uint32 atoms_of_element_56 = 1056;
    uint32 atoms_of_element_57 = 1057;
    uint32 atoms_of_element_58 = 1058;
    uint32 atoms_of_element_59 = 1059;
    uint32 atoms_of_element_60 = 1060;
    uint32 atoms_of_element_61 = 1061;
    uint32 atoms_of_element_62 = 1062;
    uint32 atoms_of_element_63 = 1063;
    uint32 atoms_of_element_64 = 1064;
    uint32 atoms_of_element_65 = 1065;
    uint32 atoms_of_element_66 = 1066;
    uint32 atoms_of_element_67 = 1067;
    uint32 atoms_of_element_68 = 1068;
    uint32 atoms_of_element_69 = 1069;
    uint32 atoms_of_element_70 = 1070;
    uint32 atoms_of_element_71 = 1071;
    uint32 atoms_of_element_72 = 1072;
    uint32 atoms_of_element_73 = 1073;
    uint32 atoms_of_element_74 = 1074;
    uint32 atoms_of_element_75 = 1075;
    uint32 atoms_of_element_76 = 1076;
    uint32 atoms_of_element_77 = 1077;
    uint32 atoms_of_element_78 = 1078;
    uint32 atoms_of_element_79 = 1079;
    uint32 atoms_of_element_80 = 1080;
    uint32 atoms_of_element_81 = 1081;
    uint32 atoms_of_element_82 = 1082;
    uint32 atoms_of_element_83 = 1083;
    uint32 atoms_of_element_84 = 1084;
    uint32 atoms_of_element_85 = 1085;
    uint32 atoms_of_element_86 = 1086;
    uint32 atoms_of_element_87 = 1087;
    uint32 atoms_of_element_88 = 1088;
    uint32 atoms_of_element_89 = 1089;
    uint32 atoms_of_element_90 = 1090;
    uint32 atoms_of_element_91 = 1091;
    uint32 atoms_of_element_92 = 1092;
    uint32 atoms_of_element_93 = 1093;
    uint32 atoms_of_element_94 = 1094;
    uint32 atoms_of_element_95 = 1095;
    uint32 atoms_of_element_96 = 1096;
    uint32 atoms_of_element_97 = 1097;
    uint32 atoms_of_element_98 = 1098;
    uint32 atoms_of_element_99 = 1099;
    uint32 atoms_of_element_100 = 1100;
    uint32 atoms_of_element_101 = 1101;
    uint32 atoms_of_element_102 = 1102;
    uint32 atoms_of_element_103 = 1103;
    uint32 atoms_of_element_104 = 1104;
    uint32 atoms_of_element_105 = 1105;
    uint32 atoms_of_element_106 = 1106;
    uint32 atoms_of_element_107 = 1107;
    uint32 atoms_of_element_108 = 1108;
    uint32 atoms_of_element_109 = 1109;
    uint32 atoms_of_element_110 = 1110;
    uint32 atoms_of_element_111 = 1111;
    uint32 atoms_of_element_112 = 1112;
    uint32 atoms_of_element_113 = 1113;
    uint32 atoms_of_element_114 = 1114;
    uint32 atoms_of_element_115 = 1115;
    uint32 atoms_of_element_116 = 1116;
    uint32 atoms_of_element_117 = 1117;
    uint32 atoms_of_element_118 = 1118;
}
